blender/arm/material/make_mesh.py

```diff
@@ -692,6 +708,15 @@ def make_forward_base(con_mesh, parse_opacity=False, transluc_pass=False):
         frag.write('}')
     frag.write('vec3 direct = vec3(0.0);')
 
+    is_shadeless = mat_state.emission_type == mat_state.EmissionType.SHADELESS
```

Comment:
materialID is read here and we set matid.
The reason is that make_forward_base is called before any other setups in make_refract and the matid is used for sampling light in make_forward_base (for transluc too)